src/napari_spatialdata/_view.py

```diff
@@ -19,12 +19,203 @@
+        self.color_widget = ScatterListWidget(self.viewer, self.model, attr="obsm")
+        self.color_widget.setAttribute("obs")
+
+        self.color_component_widget = ComponentWidget(self.model, attr="obs")
+        self.color_component_widget.setToolTip("obs")
+        self.color_component_widget.currentTextChanged.connect(self.color_widget.setComponent)
+        self.color_widget.itemClicked.connect(self.color_component_widget._onClickChange)
+
+        self.layout().addWidget(color_label)
+        self.layout().addWidget(self.color_widget)
+        self.layout().addWidget(self.color_component_widget)
+
+        self.color_selection_widget.currentTextChanged.connect(self.color_widget.setAttribute)
+        self.color_selection_widget.currentTextChanged.connect(self.color_component_widget.setAttribute)
+        self.color_selection_widget.currentTextChanged.connect(self.color_component_widget.setToolTip)
+
+        self.plot_button_widget = QPushButton("Plot")
+        self.plot_button_widget.clicked.connect(
+            lambda: self.matplotlib_widget._onClick(
+                self.x_widget.getData(), self.x_widget.text, self.y_widget.getData(), self.y_widget.text
+            )
+        )
+
+        self.layout().addWidget(self.plot_button_widget)
+
+        self.model.events.adata.connect(self._on_selection)
+
+    def _on_selection(self, event: Optional[Any] = None) -> None:
+        self.x_widget.clear()
+        self.y_widget.clear()
+        self.color_widget.clear()
+        self.x_widget._onChange()
+        self.x_component_widget._onChange()
+        self.y_widget._onChange()
+        self.y_component_widget._onChange()
+        self.color_widget._onChange()
+        self.color_component_widget._onChange()
+
+    def _select_layer(self, layer: Layer) -> None:
+        """Napari layers."""
+        self.model.layer = layer
+        # if layer is not None and "adata" in layer.metadata:
+        self.model.adata = layer.metadata["adata"]
+        self.model.library_id = layer.metadata["library_id"]
+        self.model.scale = self.model.adata.uns[Key.uns.spatial][self.model.library_id][Key.uns.scalefactor_key][
+            self.model.scale_key
+        ]
+        self.model.coordinates = np.insert(
+            self.model.adata.obsm[Key.obsm.spatial][:, ::-1][:, :2] * self.model.scale, 0, values=0, axis=1
+        )
+        if "points" in layer.metadata:
+            self.model.points_coordinates = layer.metadata["points"].X
+            self.model.points_var = layer.metadata["points"].obs["gene"]
+            self.model.point_diameter = np.array([0.0] + [layer.metadata["point_diameter"]] * 2) * self.model.scale
+        self.model.spot_diameter = (
+            np.array([0.0] + [Key.uns.spot_diameter(self.model.adata, Key.obsm.spatial, self.model.library_id)] * 2)
+            * self.model.scale
+        )
+        self.model.labels_key = layer.metadata["labels_key"] if isinstance(layer, Labels) else None
```

Comment:
why would you need this? I don't think you need to plot the stuff in the scatter back in the spatial coordinate right? better to clean up